<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtMDI</class>
 <widget class="QMainWindow" name="QtMDI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>429</width>
    <height>857</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtMDI</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabEdit">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabGCode">
       <attribute name="title">
        <string>G Code</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="CodeEditor" name="te_Code"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>429</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionRecentFiles"/>
    <addaction name="actionLoadLast"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAboutQtMDI"/>
    <addaction name="actionAboutQt"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="floatable">
    <bool>true</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="separator"/>
   <addaction name="actionRunProgram"/>
   <addaction name="actionPauseProgram"/>
   <addaction name="actionStep"/>
   <addaction name="actionStopProgram"/>
   <addaction name="separator"/>
   <addaction name="actionAboutQtMDI"/>
   <addaction name="actionAboutQt"/>
  </widget>
  <action name="actionExit">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/exit.png</normaloff>:/new/CNCicons/icons/exit.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/fileopen.png</normaloff>:/new/CNCicons/icons/fileopen.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open (Ctrl+0)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionRecentFiles">
   <property name="text">
    <string>Recent Files</string>
   </property>
  </action>
  <action name="actionEdit">
   <property name="text">
    <string>Edit</string>
   </property>
  </action>
  <action name="actionReload">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/reload.png</normaloff>:/new/CNCicons/icons/reload.png</iconset>
   </property>
   <property name="text">
    <string>Reload</string>
   </property>
   <property name="toolTip">
    <string>Reload (CTRL+R)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionProperties">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/properties.png</normaloff>:/new/CNCicons/icons/properties.png</iconset>
   </property>
   <property name="text">
    <string>Properties</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/filesaveas.png</normaloff>:/new/CNCicons/icons/filesaveas.png</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="toolTip">
    <string>Save As (Ctrl + Shift + S)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionToolTable">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/edit.png</normaloff>:/new/CNCicons/icons/edit.png</iconset>
   </property>
   <property name="text">
    <string>Edit Tool Table</string>
   </property>
   <property name="toolTip">
    <string>Edit Tool Table</string>
   </property>
  </action>
  <action name="actionReloadToolTable">
   <property name="text">
    <string>Reload Tool Table</string>
   </property>
  </action>
  <action name="actionToggleEstop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/estop.png</normaloff>:/new/CNCicons/icons/estop.png</iconset>
   </property>
   <property name="text">
    <string>Toggle ESTOP</string>
   </property>
   <property name="toolTip">
    <string>Toggle ESTOP (F1)</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionToggleMachineOn">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/machine_toggle.png</normaloff>:/new/CNCicons/icons/machine_toggle.png</iconset>
   </property>
   <property name="text">
    <string>Toggle Machine ON</string>
   </property>
   <property name="toolTip">
    <string>Toggle Machine ON (F2)</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionRunProgram">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/run_program.png</normaloff>:/new/CNCicons/icons/run_program.png</iconset>
   </property>
   <property name="text">
    <string>Run Program</string>
   </property>
   <property name="toolTip">
    <string>Run Program (R)</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionPauseProgram">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/pause_program.png</normaloff>:/new/CNCicons/icons/pause_program.png</iconset>
   </property>
   <property name="text">
    <string>Pause Program</string>
   </property>
   <property name="toolTip">
    <string>Pause Program (Space)</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionStopProgram">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/stop_program.png</normaloff>:/new/CNCicons/icons/stop_program.png</iconset>
   </property>
   <property name="text">
    <string>Stop Program</string>
   </property>
   <property name="toolTip">
    <string>Stop Program (Esc)</string>
   </property>
   <property name="shortcut">
    <string>Esc</string>
   </property>
  </action>
  <action name="actionStep">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/skip_program.png</normaloff>:/new/CNCicons/icons/skip_program.png</iconset>
   </property>
   <property name="text">
    <string>Step</string>
   </property>
   <property name="toolTip">
    <string>Step (S)</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionAboutQtMDI">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/help_about.png</normaloff>:/new/CNCicons/icons/help_about.png</iconset>
   </property>
   <property name="text">
    <string>About QtMDI</string>
   </property>
  </action>
  <action name="actionAboutQt">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/Qt.png</normaloff>:/new/CNCicons/icons/Qt.png</iconset>
   </property>
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionHome">
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/kfm_home.png</normaloff>:/new/CNCicons/icons/kfm_home.png</iconset>
   </property>
   <property name="text">
    <string>Home</string>
   </property>
   <property name="toolTip">
    <string>Home (Home)</string>
   </property>
   <property name="shortcut">
    <string>Home</string>
   </property>
  </action>
  <action name="actionLoadLast">
   <property name="text">
    <string>Load Last File</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionOptionalStop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../qtMDI/qtMDI.qrc">
     <normaloff>:/new/CNCicons/icons/optional.png</normaloff>:/new/CNCicons/icons/optional.png</iconset>
   </property>
   <property name="text">
    <string>Optional Stop</string>
   </property>
   <property name="toolTip">
    <string>Optional Stop</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Run</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>code_editor.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../qtMDI/qtMDI.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>onAxis(bool)</slot>
  <slot>onHome()</slot>
  <slot>onTouchOff()</slot>
  <slot>onJogMinus()</slot>
  <slot>onJogPlus()</slot>
  <slot>onCbJogMethod(int)</slot>
  <slot>onTabViewChanged(int)</slot>
  <slot>onTabControlChanged(int)</slot>
  <slot>onReturnMDICommand()</slot>
  <slot>onSelectFromMDIList(QListWidgetItem*)</slot>
  <slot>onMdiClear()</slot>
  <slot>onMdiCopy()</slot>
  <slot>onMdiPaste()</slot>
  <slot>onEdit()</slot>
  <slot>onGCodeChanged()</slot>
  <slot>onAbout()</slot>
  <slot>onJogOff()</slot>
  <slot>onJogSliderChanged(int)</slot>
  <slot>onEditToolTable()</slot>
  <slot>onFeedSliderChanged(int)</slot>
  <slot>onVelocitySliderChanged(int)</slot>
  <slot>onUnReHome()</slot>
  <slot>onResetDiameter()</slot>
 </slots>
</ui>
